### PR TITLE
Troubleshooting guide: Add help link to change encoding scheme #7274

### LINKED ISSUE #7274: Troubleshooting guide: Add help link to change encoding scheme
Quite a few people have been facing issues trying to get the encoding scheme right, despite having all the project specific properties as per our `ide-usage.md`. 

[This link](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/) seems to have solved their issues successfully. Add this to the `troubleshooting-guide.md` under `Common setup errors and solutions`

### PR DESCRIPTION
**Pull request by RaghavBatra**
Fixes #7274 

**Outline of Solution**

Rewrote the problem stated in the discuss, along with the reason and solution.

### CODE REVIEW

**Comment by RaghavBatra** [comment]
@VamsiSangam here!

File: docs/troubleshooting-guide.md [hunk 1]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+  
+* **ERROR**: Some characters are displayed incorrectly in browser. Example, Charlés's is displayed as CharlÃ©s.
```
Comment (VamsiSangam) [inline docs/troubleshooting-guide.md]: Wrap the words `Charlés's` and `CharlÃ©s` in `

**Comment by RaghavBatra** [comment]
@VamsiSangam done!

File: docs/troubleshooting-guide.md [hunk 2]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+  
+* **ERROR**: Some characters are displayed incorrectly in browser. Example, `Charlés's` is displayed as `CharlÃ©s`.
+
+  **REASON**: Page encoding is not set to UTF-8.
+
+  **SOLUTION**: Go to `Eclipse` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If you are a Windows user, then you can navigate to `Preferences` by `Window` → `Pereferences`. If this change does not solve the error, then you can try the methods in [this](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/) link.
```
Comment (VamsiSangam) [inline docs/troubleshooting-guide.md]: Change 

> If this change does not solve the error, then you can

to

> If this does not fix the error, you can

**Comment by RaghavBatra** [comment]
Done!

File: docs/troubleshooting-guide.md [hunk 3]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+  
+* **ERROR**: Some characters are displayed incorrectly in browser. Example, `Charlés's` is displayed as `CharlÃ©s`.
```
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: ...in **the** browser. **For** example, ...

File: docs/troubleshooting-guide.md [hunk 4]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+
```
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: Delete the trailing space.

File: docs/troubleshooting-guide.md [hunk 5]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+  
+* **ERROR**: Some characters are displayed incorrectly in browser. Example, `Charlés's` is displayed as `CharlÃ©s`.
+
+  **REASON**: Page encoding is not set to UTF-8.
+
+  **SOLUTION**: Go to `Eclipse` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If you are a Windows user, then you can navigate to `Preferences` by `Window` → `Pereferences`. If this does not fix the error, you can try the methods in [this](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/) link.
+
```
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: Delete this blank line.

File: docs/troubleshooting-guide.md [hunk 6]
```diff
@@ -46,6 +46,13 @@ Note that some of the screenshots might be outdated, but the instructions will r
 * **ERROR**: Eclipse complains "NewClass cannot be resolved to a type", "The import some.package.NewClass cannot be resolved", or "The method someMethod() from the type ExistingClass refers to the missing type NewClass" after syncing with `master` branch.
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
+  
+* **ERROR**: Some characters are displayed incorrectly in browser. Example, `Charlés's` is displayed as `CharlÃ©s`.
+
+  **REASON**: Page encoding is not set to UTF-8.
+
+  **SOLUTION**: Go to `Eclipse` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If you are a Windows user, then you can navigate to `Preferences` by `Window` → `Pereferences`. If this does not fix the error, you can try the methods in [this](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/) link.
```
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: For consistency with the first error, use `Window` → `Preferences` in the first sentence and delete the second sentence (which contains a typo). Place `link` within the linked part i.e. `[this link](...).`.

**Comment by RaghavBatra** [comment]
@whipermr5 @VamsiSangam done!

File: docs/troubleshooting-guide.md [hunk 7]
```diff
@@ -47,12 +47,11 @@ Note that some of the screenshots might be outdated, but the instructions will r
...
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
   
-* **ERROR**: Some characters are displayed incorrectly in browser. Example, `Charlés's` is displayed as `CharlÃ©s`.
+* **ERROR**: Some characters are displayed incorrectly in the browser. For example, `Charlés's` is displayed as `CharlÃ©s`.
 
   **REASON**: Page encoding is not set to UTF-8.
 
-  **SOLUTION**: Go to `Eclipse` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If you are a Windows user, then you can navigate to `Preferences` by `Window` → `Pereferences`. If this does not fix the error, you can try the methods in [this](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/) link.
-
+  **SOLUTION**: Go to `Windows` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If this does not fix the error, you can try the methods in [this link](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/).
```
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: ~~Windows~~ Window

File: docs/troubleshooting-guide.md (hunk 4 above)
Comment (whipermr5) [inline docs/troubleshooting-guide.md]: The trailing spaces are still there.

**Comment by RaghavBatra** [comment]
@whipermr5 is this OK?
If not, can you tell me how to go about doing so?

File: docs/troubleshooting-guide.md [hunk 8]
```diff
@@ -47,6 +47,12 @@ Note that some of the screenshots might be outdated, but the instructions will r
 
   **SOLUTION**: This is likely because the dependencies have changed. Refer to [this document](dependencies.md) for steps to update your local dependencies configuration.
 
+* **ERROR**: Some characters are displayed incorrectly in the browser. For example, `Charlés's` is displayed as `CharlÃ©s`.
+
+  **REASON**: Page encoding is not set to UTF-8.
+
+  **SOLUTION**: Go to `Window` → `Preferences` → `Resources` → change the `Text file encoding` setting from `Default` to `Other: UTF-8`. If this does not fix the error, you can try the methods in [this link](https://z0ltan.wordpress.com/2011/12/25/changing-the-encoding-in-eclipse-to-utf-8-howto/).
+
```
Comment (damithc) [inline docs/troubleshooting-guide.md]: Is this specific to Eclipse? There is no mention of Eclipse in the text.

Comment (whipermr5) [inline docs/troubleshooting-guide.md]: @RaghavBatra Do add a reference to Eclipse in the text.

**Comment by RaghavBatra** [comment]
Done!

**Comment by RaghavBatra** [comment]
@damithc @whipermr5

**Pull request merged**